x-pack/plugins/endpoint/public/applications/endpoint/store/policy_details/reducer.ts

```diff
@@ -93,10 +95,19 @@ export const policyDetailsReducer: Reducer<PolicyDetailsState, AppAction> = (
       return state;
     }
     const newState = { ...state, policyItem: { ...state.policyItem } };
-    const newPolicy: any = { ...fullPolicy(state) };
+    const newPolicy: PolicyConfig = { ...fullPolicy(state) };
+
+    /**
+     * This is directly changing redux state because `policyItem.inputs` was copied over and not cloned.
```

Comment:
not 100% sure that this is mutating redux state, but im pretty sure its not safe

Reply:
yeah, you are right here. Missed that. I'm thinking that we should just start to use lodash `cloneDeep` especially when working with these deep object structures.